tcc/src/main/java/io/seata/rm/tcc/api/BusinessActionContext.java

```diff
@@ -145,11 +145,6 @@ public void setActionName(String actionName) {
 
     @Override
     public String toString() {
-        StringBuilder sb = new StringBuilder();
-        sb.append("[xid:").append(xid)
-            .append(",branch_Id:").append(branchId).append(",action_name:").append(actionName)
-            .append(",action_context:")
-            .append(actionContext).append("]");
-        return sb.toString();
+        return "[xid:" + xid + ",branch_Id:" + branchId + ",action_name:" + actionName + ",action_context:" + actionContext + "]";
```

Comment:
?

Reply:
I think the use of StringBuilder is generally in a loop. In fact, simple string splicing will be optimized by JVM to StringBuilder. Personally, the efficiency is not much different, and the code is more concise. [Stackoverflow](https://stackoverflow.com/questions/1532461/stringbuilder-vs-string-concatenation-in-tostring-in-java) has related questions and answers on this question.

Reply:
@lzf971107 will there be more memory fragmentation here?

Reply:
> @lzf971107 will there be more memory fragmentation here?

They have the same code after decompilation, and the effect they cause is actually the same. If memory fragmentation occurs, then StringBuilder is also the same.